HW3/HW3_Khyzhniak.js

```diff
@@ -0,0 +1,133 @@
+function task1() {
+  console.log('%c --- TASK #1 --- ', 'color: green; font-weight: bold; font-size: 1.5em;');
+
+  function sum(a = 0) {
+    let sum = a;
+    
+    return a === 0 ? a :
+    function add(b) {
+      return b === undefined ? sum : (
+        sum += b,
+        add
+      )
+    }
+  }
+  
+  console.log(sum()); // 0
+  console.log(sum(3)()); // 3
+  console.log(sum(1)(3)(7)()); // 11
+  console.log(sum(1)(3)(7)(8)()); // 19
+  
+  console.log('%c --- OPTIONAL --- ', 'color: green; font-weight: bold; font-size: 1.5em;');
+
+  function calcSum(sum) {
+    function add(b) {
+      return b === undefined ? sum :
+      (    
+      sum += b,
+      add
+      )
+    }
+    
+    return sum === undefined ? 0 : add;
+  }
+
+  console.log(calcSum()); // 0
+  console.log(calcSum(3)()); // 3
+  console.log(calcSum(1)(3)(7)()); // 11
+  console.log(calcSum(1)(3)(7)(8)()); // 19
+
+  /* FOR BETTER READABILITY:
+  function sum(acc) {
+    function add(num) {
+      if (num === undefined) return acc;
+
+      acc += num;
+      return add;
+    }
+
+    return acc === undefined ? 0 : add;
+  }
+
+  console.log(sum()); // 0
+  console.log(sum(3)()); // 3
+  console.log(sum(1)(3)(7)()); // 11
+  console.log(sum(1)(3)(7)(8)()); // 19
+  */
+};
+task1();
+
+function task2() {
+  console.log('%c --- TASK #2 --- ', 'color: green; font-weight: bold; font-size: 1.5em;');
+
+  const treeRoot = {
+    data: 0,
+    parent: null,
+    children: []
+  };
+
+  const treeFirstLevel1 = {
+    data: 10,
+    parent: treeRoot,
+    children: []
+  };
+  treeRoot.children.push(treeFirstLevel1);
+
+  const treeFirstLevel2 = {
+    data: 5,
+    parent: treeRoot,
+    children: []
+  };
+  treeRoot.children.push(treeFirstLevel2);
```

Comment:
why do you do additional calculation operations instead of describing the whole tree at once?

Reply:
That's from an example. I was following the instructions.